### PR TITLE
DolphinWX: Fixes for compressing multiple games at once

### PR DESCRIPTION
**Pull request by JosJuice**
Read the commits for details.

[Review on Reviewable](https://reviewable.io/reviews/dolphin-emu/dolphin/3347)

### CODE REVIEW

File: Source/Core/DolphinWX/GameListCtrl.cpp [hunk 1]
```diff
@@ -949,20 +959,23 @@ const GameListItem * CGameListCtrl::GetSelectedISO()
...
+std::vector<const GameListItem*> CGameListCtrl::GetAllSelectedISOs() const
+{
+	std::vector<const GameListItem*> result;
+	long item = -1;
+	while (true)
+	{
+		item = GetNextItem(item, wxLIST_NEXT_ALL, wxLIST_STATE_SELECTED);
+		if (item == wxNOT_FOUND)
+			return result;
+		else
```
Comment (lioncash) [inline Source/Core/DolphinWX/GameListCtrl.cpp]: else isn't necessary here.

File: Source/Core/DolphinWX/GameListCtrl.cpp [hunk 2]
```diff
@@ -69,9 +69,19 @@
 #include "DolphinWX/resources/Platform_Wii.xpm"
 #include "DolphinWX/resources/rating_gamelist.h"
 
-size_t CGameListCtrl::m_currentItem = 0;
-size_t CGameListCtrl::m_numberItem = 0;
-std::string CGameListCtrl::m_currentFilename;
+struct CompressionProgress
```
Comment (lioncash) [inline Source/Core/DolphinWX/GameListCtrl.cpp]: struct should be marked as final

**Comment by aserna3** [comment]
Tested with two different Wii games and a GC game. The warning only shows once now.

**Comment by JosJuice** [comment]
The Wii warning appearing more than once is a regression (we used to not have a warning at all), so I'm marking this for 5.0. If this doesn't make it in before the feature freeze, I might have to throw out the last two commits.

File: Source/Core/DolphinWX/GameListCtrl.cpp [hunk 3]
```diff
@@ -69,9 +69,19 @@
...
-size_t CGameListCtrl::m_numberItem = 0;
-std::string CGameListCtrl::m_currentFilename;
+struct CompressionProgress final
+{
+public:
+	CompressionProgress(int items_done, int items_total, const std::string& current_filename, wxProgressDialog& dialog)
+		: items_done(items_done), items_total(items_total), current_filename(current_filename), dialog(dialog)
+	{ }
+
+	int items_done;
```
Comment (Sonicadvance1) [inline Source/Core/DolphinWX/GameListCtrl.cpp]: These aren't following member naming convention

Comment (JosJuice) [inline Source/Core/DolphinWX/GameListCtrl.cpp]: I thought that only was a rule for classes?

Comment (JosJuice) [inline Source/Core/DolphinWX/GameListCtrl.cpp]: Actually, never mind, I need to change either this or the constructor variables to not have the same names as each other anyway. I'll change these to m_.

Comment (lioncash) [inline Source/Core/DolphinWX/GameListCtrl.cpp]: I'll need to clarify the coding style with regards to this later tonight, I meant to address this a while ago.

m_ shouldn't be a requirement for dumb/POD objects

Comment (lioncash) [inline Source/Core/DolphinWX/GameListCtrl.cpp]: You can just add a `_` at the end of the constructor items, if you prefer non-`m_` notation for these.

Comment (JosJuice) [inline Source/Core/DolphinWX/GameListCtrl.cpp]: Should be good to go now.

**Comment by JosJuice** [comment]
Ping. Should I remove the last two commits from this PR, or are they okay to have in 5.0 despite them not being fixes for regressions from 4.0?

**Comment by delroth** [comment]
I'm fine with them in 5.0.

**Comment by delroth** [comment]
Reviewed 2 of 2 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[Source/Core/DolphinWX/GameListCtrl.cpp, line 1062 \[r1\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3347#-K8sC11hdK20e-Z7jfdo:-K8sC11hdK20e-Z7jfdp:-381340204) ([raw file](https://github.com/dolphin-emu/dolphin/blob/470a2783a6e0cc2f6208d5ce72cd3f9e315e96b5/Source/Core/DolphinWX/GameListCtrl.cpp#L1062)):*
reinterpret_cast<CompressionProgress*>

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3347)*

**Comment by BhaaLseN** [comment]
Is there anything (else) blocking this from being merged?

---

Reviewed 2 of 2 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[Source/Core/DolphinWX/GameListCtrl.cpp, line 1062 \[r1\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3347#-K8sC11hdK20e-Z7jfdo:-KA0Ajsxghkt0Tin2WR0:289993079) ([raw file](https://github.com/dolphin-emu/dolphin/blob/470a2783a6e0cc2f6208d5ce72cd3f9e315e96b5/Source/Core/DolphinWX/GameListCtrl.cpp#L1062)):*
Hmm, shouldn't a `static_cast` be good enough here to go from `void*` to `CompressionProgress*`? I kinda have `reinterpret_cast` as last-resort type of thing in the back of my mind...

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3347#-:-KA0BxOSULB27Is6OaWh:-2112194790)*

**Comment by JosJuice** [comment]
There's nothing blocking this as far as I know.

**Comment by BhaaLseN** [comment]
Reviewed 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3347)*

**Comment by lioncash** [comment]
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[a discussion](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3347#-KAHCCkjAqLWRl_kUjBm:-KAHCCkklS4TZrN0cdgw:-318625647) (no related file):*
Aside from that minor nitpick, feel free to merge it when amended.

---

*[Source/Core/DolphinWX/GameListCtrl.cpp, line 959 \[r3\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3347#-KAHBlL9LIDsRfVU_-3I:-KAHBlL9LIDsRfVU_-3J:-484808058) ([raw file](https://github.com/dolphin-emu/dolphin/blob/8a2e678ea745d228c94f66ba49596e3011fcda8d/Source/Core/DolphinWX/GameListCtrl.cpp#L959)):*
the declaration of `item` can be joined with the assignment here.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3347)*

**Comment by JosJuice** [comment]
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[Source/Core/DolphinWX/GameListCtrl.cpp, line 959 \[r3\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3347#-KAHBlL9LIDsRfVU_-3I:-KAJkY1Hd1aTEq2rCOjv:2044818970) ([raw file](https://github.com/dolphin-emu/dolphin/blob/8a2e678ea745d228c94f66ba49596e3011fcda8d/Source/Core/DolphinWX/GameListCtrl.cpp#L959)):*
I need to use the value of `item` (either -1 or the value from the previous loop iteration) for the function call on this line, so it needs to be declared before this.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3347)*

**Comment by lioncash** [comment]
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[Source/Core/DolphinWX/GameListCtrl.cpp, line 959 \[r3\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3347#-KAHBlL9LIDsRfVU_-3I:-KAKUr--NaFx9DymW4TQ:2013405983) ([raw file](https://github.com/dolphin-emu/dolphin/blob/8a2e678ea745d228c94f66ba49596e3011fcda8d/Source/Core/DolphinWX/GameListCtrl.cpp#L959)):*
Whoops, missed that

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3347)*

**Comment by lioncash** [comment]
Reviewed 1 of 2 files at r2, 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3347)*

**Pull request merged**